react/week3/todolist/src/Todos.js

```diff
@@ -0,0 +1,195 @@
+  const [isEditing, setIsEditing] = useState(false);
+
+  const { description, checked, deadline, id } = todo;
+
+  function onDescriptionChange(event) {
+    const value = event.target.value;
+    updateDescription(id, value);
+  }
+
+  function onDeadlineChange(event) {
+    const value = event.target.value;
+    updateDeadline(id, value);
+  }
+
+  return (
+    <div>
+      <div style={{ textDecoration: checked ? "line-through" : "none" }}>
+        {isEditing ? (
+          <input
+            type="text"
+            value={description}
+            onChange={onDescriptionChange}
+          />
+        ) : (
+          <h3>{description}</h3>
+        )}
+        {isEditing ? (
+          <input
+            type="date"
+            id="deadline"
+            name="deadline"
+            value={deadline}
+            onChange={onDeadlineChange}
+          />
+        ) : (
+          <h4>{deadline}</h4>
+        )}
+      </div>
+      <input type="checkbox" checked={checked} onChange={onCheck} />
+      <button onClick={onDelete}>Delete</button>
+      <button onClick={() => setIsEditing(!isEditing)}>
+        {isEditing ? "Update" : "Edit"}
+      </button>
+    </div>
+  );
+}
+
+function AddTodoForm({ addTodo }) {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+
+  function onSubmit(event) {
+    event.preventDefault();
+
+    addTodo(description, deadline);
+
+    setDescription("");
+    setDeadline("");
+  }
+
+  const onDescriptionChange = (event) => {
+    const value = event.target.value;
+    setDescription(value);
+  };
+
+  const onDeadlineChange = (event) => {
+    const value = event.target.value;
+    setDeadline(value);
+  };
+
+  return (
+    <form onSubmit={onSubmit}>
+      <div>
+        <label htmlFor="description">Description</label>
+        <input
+          type="text"
+          name="description"
+          id="description"
+          value={description}
+          onChange={onDescriptionChange}
+        />
+      </div>
+      <div>
+        <label htmlFor="deadline"></label>
+        <input
+          type="date"
+          id="deadline"
+          name="deadline"
+          value={deadline}
+          onChange={onDeadlineChange}
+        />
+      </div>
+      <button type="submit">Add Todo</button>
+    </form>
+  );
+}
+
+function Todos() {
+  const [todos, setTodos] = useState([]);
+
+  useEffect(() => {
+    fetch(API_URL)
+      .then((res) => res.json())
+      .then((data) => {
+        setTodos(data);
+      });
+  }, []);
+
+  function checkTodo(id) {
+    setTodos((prevTodos) => {
+      return prevTodos.map((todo) => {
+        if (todo.id === id) {
+          return { ...todo, checked: !todo.checked };
+        } else {
+          return todo;
+        }
+      });
+    });
+  }
+
+  function addTodo(description, deadline) {
+    setTodos((prevTodos) => {
+      const newTodo = {
+        id: prevTodos.length + 1,
```

Comment:
This will give you problems when someone deletes a todo, because you can end up with equal ids.
Say you have the following todos:
```
const todos = [
  {id: 1},
  {id: 2},
  {id: 3}
] 
```
and the user deletes the second todo.
When they add a todo next, it'll get the id "3". (Length of list + 1)